### PR TITLE
e2e/verify/routes: wait for console route to exist

### PR DESCRIPTION
**Pull request by jbpratt**
it seems to not exist at first on a cluster. Also turn this into a table test!

Signed-off-by: Brady Pratt <bpratt@redhat.com>

/assign @ryankwilliams

### CODE REVIEW

File: pkg/e2e/verify/routes.go [hunk 1]
```diff
@@ -36,51 +30,45 @@ func init() {
 var _ = ginkgo.Describe(routesTestName, ginkgo.Ordered, label.E2E, func() {
```
Comment (ryankwilliams) [inline pkg/e2e/verify/routes.go:30]: @jbpratt Could we add the `labels.HyperShift`, this way these tests get included when the hypershift config gets called.

Comment (jbpratt) [reply pkg/e2e/verify/routes.go]: added!

File: pkg/e2e/verify/routes.go [hunk 2]
```diff
@@ -33,54 +27,48 @@ func init() {
...
-		err := client.Get(ctx, consoleLabel, consoleNamespace, &consoleRoute)
-		expect.NoError(err)
-
-		for _, ingress := range consoleRoute.Status.Ingress {
-			response, err := validateRoute(ctx, ingress)
-			Expect(err).NotTo(HaveOccurred(), "failed retrieving %s site", consoleRoute.Name)
-			Expect(response.StatusCode).To(Equal(http.StatusOK))
+	ginkgo.DescribeTable("exists and is operational", func(ctx context.Context, name, namespace string) {
+		if name == "oauth-openshift" && viper.GetBool(config.Hypershift) {
+			ginkgo.Skip("OAuth route is not available on a HyperShift cluster")
```
Comment (jbpratt) [inline pkg/e2e/verify/routes.go:48]: @dustman9000 do we know why there is not an oauth route?

Comment (jbpratt) [reply pkg/e2e/verify/routes.go:48]:
```
❯ oc get routes -A
NAMESPACE                             NAME                       HOST/PORT                                                                                         PATH        SERVICES                   PORT    TERMINATION          WILDCARD
open-cluster-management-agent-addon   klusterlet-addon-workmgr   klusterlet-addon-workmgr-open-cluster-management-agent-addon.apps.rw-hyper.l5tf.s3.devshift.org               klusterlet-addon-workmgr   <all>   passthrough          None
openshift-console                     console                    console-openshift-console.apps.rw-hyper.l5tf.s3.devshift.org                                                  console                    https   reencrypt/Redirect   None
openshift-console                     downloads                  downloads-openshift-console.apps.rw-hyper.l5tf.s3.devshift.org                                                downloads                  http    edge/Redirect        None
openshift-ingress-canary              canary                     canary-openshift-ingress-canary.apps.rw-hyper.l5tf.s3.devshift.org                                            ingress-canary             8080    edge/Redirect        None
openshift-monitoring                  alertmanager-main          alertmanager-main-openshift-monitoring.apps.rw-hyper.l5tf.s3.devshift.org                         /api        alertmanager-main          web     reencrypt/Redirect   None
openshift-monitoring                  prometheus-k8s             prometheus-k8s-openshift-monitoring.apps.rw-hyper.l5tf.s3.devshift.org                            /api        prometheus-k8s             web     reencrypt/Redirect   None
openshift-monitoring                  prometheus-k8s-federate    prometheus-k8s-federate-openshift-monitoring.apps.rw-hyper.l5tf.s3.devshift.org                   /federate   prometheus-k8s             web     reencrypt/Redirect   None
openshift-monitoring                  thanos-querier             thanos-querier-openshift-monitoring.apps.rw-hyper.l5tf.s3.devshift.org                            /api        thanos-querier             web     reencrypt/Redirect   None
```

File: pkg/e2e/verify/routes.go (hunk 1 above)
Comment (ryankwilliams) [reply pkg/e2e/verify/routes.go]: Thank you @jbpratt

File: pkg/e2e/verify/routes.go (hunk 2 above)
Comment (dustman9000) [reply pkg/e2e/verify/routes.go:48]: AFAIK, the oauth operator will only be running on the control plane nodes for HCP clusters, that's why you are not seeing it here.

**Comment by jbpratt** [comment]
/hold

putting this on hold. I think we need to add cluster ready checks after provisioning, before passing off to the tests.

File: pkg/e2e/verify/routes.go [hunk 3]
```diff
@@ -33,54 +27,48 @@ func init() {
...
 		expect.NoError(err)
 
-		for _, ingress := range oauthRoute.Status.Ingress {
-			response, err := validateRoute(ctx, ingress)
-			Expect(err).NotTo(HaveOccurred(), "failed retrieving %s site", oauthRoute.Name)
-			Expect(response.StatusCode).To(Equal(http.StatusForbidden))
+		for _, ingress := range route.Status.Ingress {
+			response, err := httpClient.Get(fmt.Sprintf("https://%s", ingress.Host))
+			expect.NoError(err)
+			Expect(response.StatusCode).To(Equal(http.StatusOK))
```
Comment (ryankwilliams) [inline pkg/e2e/verify/routes.go:67]: The HTTP status code differs between the console URL and OAuth URL. Console URL returns a `200` and OAuth URL returns `403`.

Can we update the `ginko.Entry` to pass in a fourth parameter for the http status and use that as part of the assertion?

```
•
------------------------------
• [FAILED] [6.789 seconds]
[Suite: e2e] Routes exists and is operational [It] OAuth [HyperShift, E2E]
/home/rywillia/osd/osde2e/pkg/e2e/verify/routes.go:72

  [FAILED] Expected
      <int>: 403
  to equal
      <int>: 200
  In [It] at: /home/rywillia/osd/osde2e/pkg/e2e/verify/routes.go:67 @ 01/13/23 12:44:38.818
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Summarizing 1 Failure:
  [FAIL] [Suite: e2e] Routes exists and is operational [It] OAuth [HyperShift, E2E]
  /home/rywillia/osd/osde2e/pkg/e2e/verify/routes.go:67
```

Comment (jbpratt) [reply pkg/e2e/verify/routes.go:67]: good catch, will update

**Comment by jbpratt** [comment]
/hold

need to check errs in wait better

**Pull request merged**